public/app/features/scenes/models/SceneComponentEditWrapper.tsx

```diff
@@ -0,0 +1,80 @@
+import { css } from '@emotion/css';
+import React, { CSSProperties } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+import { SceneObjectBase } from './SceneObjectBase';
+import { SceneComponentProps } from './types';
+
+export function SceneComponentEditWrapper<T extends SceneObjectBase<any>>({
+  model,
+  isEditing,
+}: SceneComponentProps<T>) {
+  const Component = (model as any).constructor['Component'] ?? EmptyRenderer;
+  const inner = <Component model={model} isEditing={isEditing} />;
+
+  if (!isEditing) {
+    return inner;
+  }
+
+  return <SceneComponentEditingWrapper model={model}>{inner}</SceneComponentEditingWrapper>;
+}
```

Comment:
This bit is pretty elegant. This component wraps all SceneObject react components. When in non edit mode it adds nothing. But when isEditing is will wrap the component in SceneComponentEditingWrapper which handles the focus & selection state (and adds some padding). This component can be extended to support some drag and drop & resizing capabilities.